### PR TITLE
Backend for Image Upload from Dall-E to Cloudinary

### PR DESCRIPTION
**Pull request by kahlamari**

### CODE REVIEW

File: backend/src/main/java/in/kahl/promptwhispers/service/CloudinaryService.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+@Service
+public class CloudinaryService {
+    private final Cloudinary cloudinary;
+
+    public CloudinaryService(Cloudinary cloudinary) {
+        this.cloudinary = cloudinary;
+    }
+
+    public String uploadImage(@NonNull String url) {
```
Comment (PaulKreft) [inline backend/src/main/java/in/kahl/promptwhispers/service/CloudinaryService.java:19]: Wird hier ggf. eine NullPoinertException geworfen? Und solltest du die catchen?

File: backend/src/main/java/in/kahl/promptwhispers/service/CloudinaryService.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        try {
+            var cloudinaryResponse = cloudinary.uploader().upload(url, Map.of(
+                    "resource_type", "auto",
+                    "public_id", UUID.randomUUID().toString(),
+                    "folder", "promptwhispers_test"
```
Comment (PaulKreft) [inline backend/src/main/java/in/kahl/promptwhispers/service/CloudinaryService.java:24]: soll der noch ..._test heißen?

File: backend/src/main/resources/application.properties [hunk 3]
```diff
@@ -1,4 +1,5 @@
 spring.data.mongodb.uri=${MONGODB_URI}
 app.dalle.api.url=https://api.openai.com/v1/images
 app.openai.api.key=${APP_OPENAI_API_KEY}
-app.openai.api.org=${APP_OPENAI_API_ORG}
\ No newline at end of file
+app.openai.api.org=${APP_OPENAI_API_ORG}
+CLOUDINARY_URL=${CLOUDINARY_URL}
```
Comment (PaulKreft) [inline backend/src/main/resources/application.properties:5]: Ich kenne mich leider mit Cloudinary nicht aus...

Was mich aber interessiert hat: Wieso definierst du die property hier so als Constant? Sucht Cloudinary sie sich so in diesem Format?

File: backend/src/test/java/in/kahl/promptwhispers/service/CloudinaryServiceTest.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+        when(mockCloudinary.uploader()).thenReturn(mock(Uploader.class));
+        when(mockCloudinary.uploader().upload(anyString(), anyMap())).thenReturn(mockResponse);
+
+        // ACT
+        String imageUrlActual = serviceUnderTest.uploadImage(inputUrl);
+
+        // ASSERT
+        assertEquals(mockResponse.get("secure_url"), imageUrlActual);
+    }
+}
```
Comment (PaulKreft) [inline backend/src/test/java/in/kahl/promptwhispers/service/CloudinaryServiceTest.java:43]: Hier könntest du natürlich noch `null` als Edge Case abbilden. Besonders wegen @NonNull

**Review by PaulKreft — APPROVED** [review]
Traum

File: backend/src/main/java/in/kahl/promptwhispers/service/CloudinaryService.java (hunk 1 above)
Comment (kahlamari) [reply backend/src/main/java/in/kahl/promptwhispers/service/CloudinaryService.java:19]: Jo. Genau, die Annotation stellt sicher, dass der Wert nicht Null ist. Falls er es ist, wird einen Exception geworfen. Ich muss mir mal einen Exception Handler bauen :/

File: backend/src/main/resources/application.properties (hunk 3 above)
Comment (kahlamari) [reply backend/src/main/resources/application.properties:5]: Ja. In der URL sind Key und Secret hinterlegt.

File: backend/src/test/java/in/kahl/promptwhispers/service/CloudinaryServiceTest.java (hunk 4 above)
Comment (kahlamari) [reply backend/src/test/java/in/kahl/promptwhispers/service/CloudinaryServiceTest.java:43]: Habe ich mir als Issue aufgeschrieben.
